test/e2e/kubectl.go

```diff
@@ -159,11 +162,40 @@ var _ = Describe("Kubectl client", func() {
 		It("should support exec", func() {
 			By("executing a command in the container")
 			execOutput := runKubectl("exec", fmt.Sprintf("--namespace=%v", ns), simplePodName, "echo", "running", "in", "container")
-			expectedExecOutput := "running in container"
-			if execOutput != expectedExecOutput {
-				Failf("Unexpected kubectl exec output. Wanted '%s', got '%s'", execOutput, expectedExecOutput)
+			if e, a := "running in container", execOutput; e != a {
+				Failf("Unexpected kubectl exec output. Wanted %q, got %q", e, a)
+			}
+
+			By("executing a command in the container with noninteractive stdin")
+			execOutput = newKubectlCommand("exec", fmt.Sprintf("--namespace=%v", ns), "-i", simplePodName, "cat").
+				withStdinData("abcd1234").
+				exec()
+			if e, a := "abcd1234", execOutput; e != a {
+				Failf("Unexpected kubectl exec output. Wanted %q, got %q", e, a)
+			}
+
+			// pretend that we're a user in an interactive shell
+			br := newBlockingReader("echo hi\nexit\n")
+			// make sure to close the blocking reader so the copy from br to w below can unblock
+			defer br.Close()
+
+			// use a pipe here so cmd.Wait() isn't waiting on the io.Copy of stdin to
+			// finish, as it won't unblock until the deferred br.Close() above is called
+			r, w, err := os.Pipe()
```

Comment:
comment this is to give a `*os.File` to the Cmd so it doesn't wrap in a blocking Pipe